instrumentation/net/http/splunkhttp/handler.go

```diff
@@ -0,0 +1,89 @@
+// Copyright Splunk Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package splunkhttp
+
+import (
+	"net/http"
+	"os"
+	"strings"
+
+	"go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp"
+)
+
+const (
+	EnvVarServerTimingEnabled = "SPLUNK_CONTEXT_SERVER_TIMING_ENABLED"
+)
+
+// NewHandler wraps the passed handler, functioning like Splunk specific http middleware middleware,
+// in a span named after the operation and with any provided Options.
+func NewHandler(handler http.Handler, operation string, opts ...Option) http.Handler {
+	cfg := newConfig(opts...)
+
+	if cfg.ServerTimingEnabled {
+		handler = ServerTimingMiddleware(handler)
+	}
+
+	handler = otelhttp.NewHandler(handler, "server", cfg.OtelOpts...)
+	return handler
+}
+
+// WithOtelOpts is use to pass the OpenTelemetry SDK options.
+func WithOtelOpts(opts ...otelhttp.Option) Option {
+	return OptionFunc(func(cfg *config) {
+		cfg.OtelOpts = opts
+	})
+}
+
+// WithServerTiming enabled or disabled the ServerTimingMiddleware.
+func WithServerTiming(v bool) Option {
+	return OptionFunc(func(cfg *config) {
+		cfg.ServerTimingEnabled = v
+	})
+}
+
+// Option Interface used for setting *optional* config properties
+type Option interface {
+	Apply(*config)
+}
+
+// config represents the available configuration options.
+type config struct {
+	OtelOpts            []otelhttp.Option
+	ServerTimingEnabled bool
```

Comment:
do we want to keep these options as public considering we have the `With...` helper methods. It's easier to make them public later but impossible to make them private without major version change :)

Reply:
Struct is not exported so it is not public API

Reply:
So why not keep fields lower case as well at least for self-documentation?

Reply:
If we would like to export it at some point then it would be easier 😉 
It is also consistent with:
- https://github.com/open-telemetry/opentelemetry-go-contrib/blob/2760096474280b3634e92431a7ed6da0744ef144/instrumentation/net/http/otelhttp/config.go#L34-L46
- https://github.com/signalfx/splunk-otel-go/blob/59c68c77a30c9c73fff5a8edcb70e4548b7535a2/distro/config.go#L34-L38
It is for me like having `public` methods in `internal` classes in .NET or `package` in Java.

@MrAlias What do you think?

Reply:
We can just update them if we want to expose this later. Imagine another dev in future exposing config struct and then unintentionally exposing all the fields because they were capitalized.

Reply:
It doesn't make too much of a difference at this point as @pellared pointed out.

One nice thing to also mention is that if we continue to unexport `config` but want to unmarshal into it in the future (i.e. yaml or json configuration) these fields will need to be exported.